File: 2021/Day-1/day-1.py
with open("input.txt", "r") as f:
    lines = f.readlines()
    numbers = [int(line.strip()) for line in lines]

def part1(numbers):
    higher = 0
    for i in range(len(numbers)):
        if (numbers[i] > numbers[i-1]):
            higher += 1
    return higher

def part2(numbers):
    higher = 0
    for i in range(len(numbers) - 2):
        if ((numbers[i] + numbers[i+1] + numbers[i+2]) > (numbers[i-1] + numbers[i] + numbers[i+1])):
            higher += 1
    return higher

if __name__ == "__main__":
    print('Part 1 solution:', part1(numbers))
    print('Part 2 solution:', part2(numbers))

File: 2021/Day-1/day-1.go
package main

import (
	"flag"
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func main() {
	flag.Parse()
	bytes, err := ioutil.ReadFile("input.txt")
	if err != nil {
		return
	}
	contents := string(bytes)
	split := strings.Split(contents, "\n")

	// part 1
	last := -1
	higher := 0
	for _, s := range split {
		i, _ := strconv.Atoi(s)
		if last != -1 && i > last {
			higher++
		}
		last = i
	}
	fmt.Println("Part 1 solution:", higher)

	// part 2
	higher = 0

	lastNums := []int{0, 0, 0}
	lastSum := 0
	for idx, s := range split {
		i, _ := strconv.Atoi(s)

		sum := lastSum + i - lastNums[0]

		lastNums = append(lastNums[1:], i)

		if idx < 3 {
			lastSum = sum
			continue
		}
		if lastSum < sum {
			higher++
		}

		lastSum = sum
	}
	fmt.Println("Part 2 solution:", higher)
}
